ADD querystring flags to force h264 and dtx

ggarber opened the pull request.

aullman commented on src/js/webrtc-intercept.js:
```diff
@@ -0,0 +1,54 @@
+(function() {
+
+  function getParameterByName(name, url) {
```
This function is repeated in ot-intercept.js let's write it once and require it in.

ggarber replied:
ok, I understood that you wanted to be able to remove it easily as soon as we had a good way to configure codecs in opentok.js so i didn't make a lot of effort on this.  Will do now.

aullman commented:
@ggarber I'm no sure if I'm not testing things correctly but I can't get H264 to work. I'm just testing locally going to localhost:5000/room?h264=true and when I use Chrome to Chrome and look in webrtc-internals I just see normal VP8. 

![image](https://cloud.githubusercontent.com/assets/733672/23534312/2773aa26-000b-11e7-8f32-9185939b452d.png)


When I do Firefox to Chrome though I get VP9??

![image](https://cloud.githubusercontent.com/assets/733672/23534297/0e121ff4-000b-11e7-9273-c04280724788.png)

Am I missing something? Did I break something in my refactor?

aullman commented:
@ggarber ok, I added a simple test that just makes sure that you can successfully connect and subscribe when you turn on h264.

ggarber commented:
@aullman 
1) It is what I tried to (badly) explain in slack.  You need to load part of the hack before opentok.js and the other part after opentok.js.  That's the reason why I had 2 scripts in webpack configuration.   I'll try to fix it.

2) H264 should work with fake devices.  Devices and codecs should be independent.    I wouldn't waste lot of time with this though.  It is very hacky no matter what and not business critical.

3) Firefox to Chrome VP9 is very surprising.  I will review that.

ggarber commented:
@aullman 

1. It was not working because I test it with Chrome Canary and you probably used Chrome and apparently the payload types have changed.  I fixed it to support both version now.

2. VP9 in Firefox was happening because your firefox version doesn't support H264 (it happens to me too if use Stable instead of Nightly).  I fixed to make the rule more strict, now it will never use VP9.

3. I think tests should work fine with fake devices.   I'm not sure how much effort we should put in this though if it is very hacky no matter what.  Maybe it is better to work first on exposing in opentok.js the APIs needed to configure the codec list and/or manipulate the SDPs properly.  But I let you decide :)

aullman commented:
Thanks @ggarber looking good. I also added the video codec to the stats drop down using the QOS event from the subscriber. I also added a quick test that verifies that the stats show the H264 codec if you pick H264. It only works in Chrome because Firefox still doesn't have the codec name in the stats.

odivorra commented on src/js/h264/modify-sdp.js:
```diff
@@ -0,0 +1,40 @@
+        pc.setRemoteDescription = function(sdp, success, failure) {
+          console.log('Intercept setRemoteDescription');
+          if (dtx) {
+            sdp.sdp = sdp.sdp.replace('useinbandfec=1', 'useinbandfec=1;usedtx=1');
```
@aullman , @ggarber  are we guys mangling this without telling the remote receiver in the offer?

odivorra replied:
can this be a problem?
